src/components/dragLineLayer.ts

```diff
@@ -0,0 +1,170 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+
+export interface DragLineCallback<D> { (dragLineLayer: Components.DragLineLayer<D>): void; };
+
+export module Components {
+  export class DragLineLayer<D> extends GuideLineLayer<D> {
+    private _dragInteraction: Interactions.Drag;
+    private _detectionRadius = 3;
+    private _detectionEdge: d3.Selection<void>;
+    private _enabled = true;
+
+    private _dragStartCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragEndCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _disconnectInteractionCallbacks: () => void;
+
+    constructor(orientation: string) {
+      super(orientation);
+      this.addClass("drag-line-layer");
+      this.addClass("enabled");
```

Comment:
Unlike `DragBoxLayer`, there aren't really other modes for `DragLineLayer`, so it seems sufficient to have enabled control the whole thing. I could use the class `movable` instead if that seems more flexible.